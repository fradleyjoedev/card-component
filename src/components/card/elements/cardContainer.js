import styled from 'styled-components'

export default styled.div`
    background-image: url('${props => props.image}');
`;