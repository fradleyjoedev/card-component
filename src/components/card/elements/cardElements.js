import styled from 'styled-components'

//Made the assumption that the image should cover the container
export const CardContainer = styled.div`
    background-image: url('${props => props.image}');
    background-size: cover;
    display: flex;
    flex-wrap: wrap;
    align-content: flex-end;
    width: 100%;
    height: 100%;
`;