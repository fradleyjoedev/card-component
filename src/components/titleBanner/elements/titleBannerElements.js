import styled from 'styled-components'
import { COLOUR_PRIMARY, COLOUR_SECONDARY } from '../../shared/globalColors';
import { MEDIAQUERY_TABLET } from '../../shared/globalMediaQueries';

export const TitleBannerContainer = styled.div`
    display: flex;
    align-items: center;
    width: 100%
    height: 20%;
    background: ${COLOUR_SECONDARY};
    padding: 2% 1%;
`;

export const SmallImage = styled.span`
    background-image: url('${props => props.image}');
    height: 100%;
    width: 20%;
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
`;

export const Heading = styled.h2`
    color: ${COLOUR_PRIMARY};
    font-size: 5.2vw;
    width: 80%;
    padding-left: 2%;

    @media (min-width: ${MEDIAQUERY_TABLET}) {
        font-size: 3.2vw;
    }
`;