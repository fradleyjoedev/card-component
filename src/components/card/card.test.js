import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components'
import Card from '.';
import { TileImage } from './elements/cardElements';

describe('Card Component', () => {
  it('should render correctly with props', () => {
    const json = {
      "image": "image",
      "banner": {
        "image": "image",
        "heading": "test"
      }
    };
    const component = shallow(<Card data={json}/>);
    
    expect(component).toMatchSnapshot();
  });

  it('should have a image', () => {
    const imageURL = "image.jpg";

    const image = mount(<TileImage src={imageURL} />);

    expect(image.find("img").prop("src")).toEqual(imageURL);
  });
});