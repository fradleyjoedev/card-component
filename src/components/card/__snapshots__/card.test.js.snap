// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
      Object {
        "banner": Object {
          "heading": "test",
          "image": "image",
        },
        "image": "image",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <GlobalStyleComponent />,
        <ForwardRef
          image="image"
        >
          <TitleBanner
            data={
              Object {
                "heading": "test",
                "image": "image",
              }
            }
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <TitleBanner
            data={
              Object {
                "heading": "test",
                "image": "image",
              }
            }
          />,
          "image": "image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "heading": "test",
              "image": "image",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
    background-image: url('",
              [Function],
              "');
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <GlobalStyleComponent />,
          <ForwardRef
            image="image"
          >
            <TitleBanner
              data={
                Object {
                  "heading": "test",
                  "image": "image",
                }
              }
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <TitleBanner
              data={
                Object {
                  "heading": "test",
                  "image": "image",
                }
              }
            />,
            "image": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Object {
                "heading": "test",
                "image": "image",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "rules": Array [
                "
    background-image: url('",
                [Function],
                "');
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
